src/cert/ssl_cert_provider.cpp

```diff
@@ -103,128 +129,216 @@ class EVPKey
     }
 
 private:
-    std::unique_ptr<EVP_PKEY, decltype(EVP_PKEY_free)*> key{EVP_PKEY_new(), EVP_PKEY_free};
+    std::unique_ptr<EVP_PKEY, decltype(&EVP_PKEY_free)> key{nullptr, EVP_PKEY_free};
 };
 
 std::vector<unsigned char> as_vector(const std::string& v)
 {
     return {v.begin(), v.end()};
 }
 
-std::string cn_name_from(const std::string& server_name)
+void set_random_serial_number(X509* cert)
 {
-    if (server_name.empty())
-        return mp::utils::make_uuid().toStdString();
```

Comment:
Was this case only ever hit in clients, before?

Reply:
OK, it might be good to add a comment next to the signature of the `SSLCertProvider` constructor. Something like: `// leave server_name empty for clients; choose a (non-empty) name for servers.`